src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -102,15 +102,27 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
         // In each lowering piece above, namespaces were tracked. We render them here to ensure every
         // lowering action has a chance to add a source location to a namespace. Ultimately, closest wins.
         var index = 0;
+
+        UsingDirectiveIntermediateNode lastDirective = null;
         foreach (var reference in usingReferences)
         {
             var @using = new UsingDirectiveIntermediateNode()
             {
                 Content = reference.Namespace,
-                Source = reference.Source,
+                Source = reference.Source
             };
 
             builder.Insert(index++, @using);
+
+            lastDirective = @using;
+        }
+
+        if (lastDirective is not null)
```

Comment:
Feels like this would be a bit cleaner if we just did `usingReferences is [.., { } lastDirective]`, rather than the assignment trick currently.

Reply:
Sadly, that doesn't work. We need the object we just inserted to `builder`, not the one in `usingReferences`. Also that object is not the last one in the builder, and the builder type can't (currently) be indexed into, in order to use the value of `index - 1` :)

Reply:
oh wait, I can do the check for the last reference inside the loop. my bad, thats probably what you meant :)

Reply:
Spoke offline, the alternative is worse in this case, so going to leave as is.